gocd/bci_repo_publish.py

```diff
@@ -147,6 +156,7 @@ def run(self, version, token=None):
         # Trigger publishing
         if token is None:
             self.logger.warning('Would publish now, but no token specified')
+            self.last_publish_mtime = packages[0]['built_mtime']
```

Comment:
why are we setting last_publish_mtime in this error case?

Reply:
It's not really an error case, more of a dry run case

Reply:
sure, but there are plenty of other returns before hand that are not setting `last_publish_mtime`. so this is a bit inconsistent.